### PR TITLE
fix(py3): Correctly handle pickle version assignment

### PR DESCRIPTION
**Pull request by evanpurkhiser**
See the added comment as to why this is needed

### CODE REVIEW

File: src/sentry/monkey/pickle.py [hunk 1]
```diff
@@ -86,12 +86,35 @@ def patch_pickle_loaders():
     #       > The protocol version of the pickle is detected automatically
     #
     # [0]: https://docs.python.org/3/library/pickle.html#pickle-protocols
+    #
+    # XXX(epurkhiser): Unfortunately changing this module property is NOT
+    # enough. Python 3 will use _pickle (aka new cpickle) if it is available
+    # (which it usually will be), in which case it will NOT read from this
+    # DEFAULT_PROTOCOL, as the module functions just pass straight to the c
+    # implementation, which does not have a mutable DEFAULT_PROTOCOL module
+    # propety.
```
Comment (wedamija) [inline src/sentry/monkey/pickle.py:95]: Nit: `property`

File: src/sentry/monkey/pickle.py [hunk 2]
```diff
@@ -127,4 +150,5 @@ def py3_compat_kombu_pickle_loads(s, load=__py3_compat_kombu_pickle_load):
         return original_kombu_pickle_loads(s, load)
 
     pickle.loads = py3_compat_pickle_loads
+    pickle.dumps = py3_compat_pickle_dumps
```
Comment (wedamija) [inline src/sentry/monkey/pickle.py:153]: Will this work with kombu as well?

File: src/sentry/monkey/pickle.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/monkey/pickle.py]: fixed ty

File: src/sentry/monkey/pickle.py (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/monkey/pickle.py:156]: I've added `kombu_serializer.pickle_protocol = 2` to handle this

**Review by wedamija — APPROVED** [review]

File: src/sentry/monkey/pickle.py [hunk 3]
```diff
@@ -86,12 +86,38 @@ def patch_pickle_loaders():
...
 
+    # Enforce protocol for kombu as well
+    kombu_serializer.pickle_protocol = 2
+
     original_pickle_load = pickle.load
     original_pickle_loads = pickle.loads
+    original_pickle_dumps = pickle.dumps
     original_kombu_pickle_loads = kombu_serializer.pickle_loads
 
+    def py3_compat_pickle_dumps(*args, **kwargs):
```
Comment (BYK) [inline src/sentry/monkey/pickle.py:108]: Wouldn't it be easier to use the PY3 signature here and move on from there:
```py
def py3_compat_pickle_dumps(obj, protocol=None, *args, **kwargs):
    return original_pickle_dumps(obj, pickle.DEFAULT_PROTOCOL, *args, **kwargs)
```

**Review by BYK — APPROVED** [review]

**Pull request merged**